perceval/converters/myqlm_converter.py

```diff
@@ -108,51 +87,19 @@ def convert(self, qlmc, use_postselection: bool = True) -> Processor:
                     gate_id = qlmc.ops[i].gate
                     gate_matrix = qlmc.gateDic[gate_id].matrix  # gate matrix data
                     gate_u = circ_to_np(gate_matrix)  # gate matrix to numpy
-                    ins = self._create_one_qubit_gate(gate_u)
-                p.add(instruction_qbit[0]*2, ins.copy())
+                    ins = super()._create_generic_1_qubit_gate(gate_u)
```

Comment:
Calling `super().__init__` in children `__init__` methods is basically what you want (i.e. call the parent initializer from the child). However, for other calls, it's not what you want to do. What you want is:
`ins = self._create_generic_1_qubit_gate(gate_u)`
which means use the `_create_generic_1_qubit_gate` method from the parent, *unless the child has overridden it*, which is good practice (that's related to the L from the SOLID principles : https://en.wikipedia.org/wiki/Liskov_substitution_principle - you have to be able to change the behavior of a system by implementing a different child class).

Reply:
Thanks a lot for this explanation :-). Fixed.